crates/bevy_asset/src/asset_server.rs

```diff
@@ -134,11 +134,13 @@ impl AssetServer {
         &self,
         extension: &str,
     ) -> Result<Arc<Box<dyn AssetLoader>>, AssetServerError> {
-        self.server
-            .extension_to_loader_index
-            .read()
-            .get(extension)
-            .map(|index| self.server.loaders.read()[*index].clone())
+        let index = {
+            // scope map to drop lock as soon as possible
+            let map = self.server.extension_to_loader_index.read();
+            map.get(extension).copied()
+        };
```

Comment:
I'm not a big fan of the block. We could just do
```rust
// scope `extension_to_loader_index` lock to prevent deadlock with `add_loader`.
let index = self.server.extension_to_loader_index.read().get(extension).copied();
```

Reply:
Hmm... I guess with the comment it might not be as likely, but my worry was some future maintainer coming along and "cleaning up" the code by splitting it into named variables, and then re-introducing the deadlock accidentally. By scoping it in a block it makes it very clear what's going on and why, IMO. Not sure if there's a best-practices doc that backs that up though, just my own hunch.

Reply:
It's not that big of a deal, It just imo the block is a little too much and just the comment would do. 
However, if you feel strongly about the block, feel free to keep as is :)